dali/sasha/saarch.cpp

```diff
@@ -1389,83 +1392,82 @@ class CCachedWorkUnitRemover: public CBranchArchiver
         }
         PROGLOG("CLEANUP: CachedWorkUnits complete (%d removed of %d)",i,n);
     }
-
-
 };
 
 
 
 
-class CSashaArchiverServer: public ISashaServer, public Thread
-{  
-
-    bool stopped;
+class CSashaArchiverServerBase : public CSimpleInterfaceOf<ISashaServer>, implements IThreaded
+{
+    CThreaded threaded;
+protected:
+    bool stopped = false;
     Semaphore stopsem;
+    Linked<IPropertyTree> archprops;
 public:
-    IMPLEMENT_IINTERFACE;
-
-    CSashaArchiverServer()
-        : Thread("CSashaArchiverServer")
+    CSashaArchiverServerBase(IPropertyTree *_config) : archprops(_config), threaded("CSashaArchiverServerBase", this)
     {
-        stopped = false;
     }
-
-    ~CSashaArchiverServer()
+    virtual void start() override
     {
+        threaded.start();
     }
-
-    void start()
+    virtual void ready() override
     {
-        Thread::start();
     }
-
-    void ready()
+    virtual void stop() override
     {
-    }
-    
-    void stop()
-    {
-        if (!stopped) {
+        if (!stopped)
+        {
             stopped = true;
             stopsem.signal();
         }
-        join();
+        threaded.join();
     }
+};
 
+#ifndef _CONTAINERIZED
```

Comment:
I think it might be better for these to not be conditionally compiled - then any compile errors would be caught in either build.